### PR TITLE
SFT Training update tutorials

### PR DESCRIPTION
**Pull request by tengomucho**
# What does this PR do?

This PR revisits the SFT Training tutorial of Llama3-8B. Few highlights of the changes:

- correct scripts, remove references to unused configurations and tools,
- update TOC,
- update wording, compile script and separate shell script to launch training,
- add model merge after consolidation, to obtain a model that can be loaded for evaluation,
- add dependencies on trl and peft.

Note that there is still an issue with the training: the loss does NOT decrease during fine-tune (cc @michaelbenayoun).

## Before submitting
- [x] This PR fixes a typo or improves the docs

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -19,6 +19,8 @@
     "huggingface_hub >= 0.20.1",
     "numpy>=1.22.2, <=1.25.2",
     "protobuf>=3.20.3, <4",
+    "trl == 0.11.4",
```
Comment (dacorvo) [inline setup.py:22]: These are the global, bare minimum dependencies: we should not add optional components here.

**Review by dacorvo — CHANGES_REQUESTED** [review]
Thansk you for the pull-request. Looks good to me except for the dependencies.

**Review by dacorvo — APPROVED** [review]

File: docs/source/training_tutorials/sft_lora_finetune_llm.mdx [hunk 2]
```diff
@@ -50,6 +56,10 @@ huggingface-cli login --token YOUR_TOKEN
 ```bash
 git clone https://github.com/huggingface/optimum-neuron.git
 ```
+5. Make sure you have the `training` extra installed, to get all the necessary dependencies:
+```bash
+python -m pip install .[training]
+```
```
Comment (michaelbenayoun) [inline docs/source/training_tutorials/sft_lora_finetune_llm.mdx:62]: Awesome!!

File: docs/source/training_tutorials/sft_lora_finetune_llm.mdx [hunk 3]
```diff
@@ -63,7 +73,7 @@ Example:
   "context": "",
   "response": (
         "World of warcraft is a massive online multi player role playing game. "
-        "It was released in 2004 by blizarre entertainment"
+        "It was released in 2004 by bizarre entertainment"
```
Comment (michaelbenayoun) [inline docs/source/training_tutorials/sft_lora_finetune_llm.mdx:76]:
```suggestion
        "It was released in 2004 by Blizzard Entertainment"
```

File: docs/source/training_tutorials/sft_lora_finetune_llm.sh [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+  --do_train \
+  --learning_rate 5e-5 \
+  --warmup_ratio 0.03 \
+  --max_steps $MAX_STEPS \
+  --per_device_train_batch_size $BS \
+  --per_device_eval_batch_size $BS \
+  --gradient_accumulation_steps $GRADIENT_ACCUMULATION_STEPS \
+  --gradient_checkpointing true \
+  --bf16 \
+  --zero_1 false \
```
Comment (michaelbenayoun) [inline docs/source/training_tutorials/sft_lora_finetune_llm.sh:40]: Can be removed

File: setup.py [hunk 5]
```diff
@@ -50,9 +50,15 @@
     "hf_doc_builder @ git+https://github.com/huggingface/doc-builder.git",
 ]
 
+TRAINING_REQUIRES = [
+    "trl == 0.11.4",
+    "peft == 0.14.0",
```
Comment (michaelbenayoun) [inline setup.py:55]: Can we add `neuronx_distributed` as well?

File: docs/source/training_tutorials/sft_lora_finetune_llm.mdx (hunk 3 above)
Comment (tengomucho) [reply docs/source/training_tutorials/sft_lora_finetune_llm.mdx:76]: Nope, that is what is actually in the Dolly dataset! See [here](https://huggingface.co/datasets/databricks/databricks-dolly-15k?views%5B%5D=train&sql_console=true&sql=SELECT+*+FROM+train+WHERE+instruction+%3D%3D+%27What+is+world+of+warcraft%27+LIMIT+10%3B) 🤷

Comment (michaelbenayoun) [reply docs/source/training_tutorials/sft_lora_finetune_llm.mdx:76]: As a former big player of WoW I feel attacked.

**Review by michaelbenayoun — APPROVED** [review]
LGTM!
That's awesome, thanks for taking care of that!
